src/Illuminate/Collections/Traits/EnumeratesValues.php

```diff
@@ -330,7 +330,7 @@ public function firstWhere($key, $operator = null, $value = null)
      */
     public function value($key, $default = null)
     {
-        if ($value = $this->firstWhere($key)) {
+        if ($value = $this->first()) {
```

Comment:
This is incorrect -- the first _non-null_ value should be retrieved, not the first in general:

```suggestion
        if ($value = $this->firstWhere($key, '!==', null)) {
```